lib/cdo/delete_accounts_helper.rb

```diff
@@ -126,18 +126,23 @@ def self.anonymize_user_sections(user_id)
   # (reporting only).
   # @param [Integer] The user ID to purge from Pardot.
   def self.remove_from_pardot(user_id)
-    pardot_ids = PEGASUS_DB[:contact_rollups].
-      select(:pardot_id).
-      where(dashboard_user_id: user_id).
-      map {|contact_rollup| contact_rollup[:pardot_id]}
-    failed_ids = Pardot.delete_prospects(pardot_ids)
-    if failed_ids.any?
-      raise "Pardot.delete_prospects failed for Pardot IDs #{failed_ids.join(', ')}."
+    # Though we have the DB tables in all environments, we only sync data from the production
+    # environment with Pardot.
+    if rack_env == :production
+      pardot_ids = PEGASUS_DB[:contact_rollups].
+        select(:pardot_id).
+        where(dashboard_user_id: user_id).
+        map {|contact_rollup| contact_rollup[:pardot_id]}
+      failed_ids = Pardot.delete_prospects(pardot_ids)
+      if failed_ids.any?
+        raise "Pardot.delete_prospects failed for Pardot IDs #{failed_ids.join(', ')}."
```

Comment:
It does make me want to do something... Let me follow-up with a change in another PR (as it is unrelated to the purpose of this PR: making Pardot interaction be robust to environment).